Razor/Core/Journal.cs

```diff
@@ -5,23 +5,76 @@
 
 namespace Assistant
 {
+    public class JournalEntry
+    {
+        public string Value { get; }
+
+        public MessageType Type { get; }
+
+        public JournalEntry(string value, MessageType type)
+        {
+            Value = value;
+            Type = type;
+        }
+    }
     public static class Journal
     {
-        private static string _content = "";
+        private static List<JournalEntry> _entries = new List<JournalEntry>();
 
         public static void AddLine(string text)
         {
-            _content += text;
+            AddLine(text, MessageType.Regular);
+        }
+
+        public static void AddLine(string text, MessageType type)
+        {
+            string newText = text;
+
+            switch (type)
+            {
+                case MessageType.System:
+                    {
+                        newText = $"System: {text}";
+                        break;
+                    }
+                case MessageType.Label:
+                    {
+                        newText = $"You see: {text}";
+                        break;
+                    }
+                default:
+                    break;
+            }
+            
+            AddLine(new JournalEntry(newText, type));
+
+            if (_entries.Count > 50)
+                _entries.RemoveAt(_entries.Count - 1);
+        }
+
+        public static void AddLine(JournalEntry entry)
+        {
+            _entries.Insert(0, entry);
```

Comment:
This is going to copy the entire journal to a new array every single time a journal message occurs. Instead of List<>, use LinkedList<>